react/components/MicInputButton.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faMicrophone,
+  faMicrophoneSlash,
+} from '@fortawesome/free-solid-svg-icons';
+
+import Recorder from '../libs/recorder';
+
+declare global {
+    interface Window { webkitAudioContext: any; }
+}
+
+interface MicInputButtonProps {
+  submit: (d: any) => void;
+}
+
+const MicInputButton: React.FC<MicInputButtonProps> = props => {
+  const [isRecording, setIsRecording] = useState(false);
+  const [stream, setStream] = useState<any>(null);
+  const [recorder, setRecorder] = useState<any>(null);
+
+  // POST request to send audio file
+  const postAudio = (blob: any) => {
+    const data = new FormData();
+    data.append('audio', blob);
+    axios({
+      method: 'post',
+      url: process.env.REACT_APP_STTURL || 'http://127.0.0.1:8000/rest/stt',
```

Comment:
Wait is this code run client-side or server-side? Shouldn't it be client-side (where you won't have process.env)?

Normally (= in other parts of the code), configuration is injected to client-side JS through data- attributes on the body, populated from the config files by the server.

Reply:
Will look into this